File: ruby/sieve/sieve.rb
class Sieve
  def initialize(num)
    @num = num
  end

  def primes(index = 2, num_list = (2..@num).to_a)
    return [] if @num < 2
    return num_list if index == @num
    new_list = num_list.map { |n| ( n == index || n % index ) == 0 ? nil : n }.compact
    primes(index +1, new_list)
  end
end

module BookKeeping
  VERSION = 3
end

File: ecmascript/sieve/sieve.js
export default class Sieve {
  constructor(n) {
    this.n = n;
  }

  get primes() {
    let list = [...Array(this.n + 1).keys()].slice(2);
    return this.calculateSieve(2, list);
  }

  calculateSieve(i, list){
    if (this.n < 2)   return [];
    if (i === this.n) return list;

    let newList = [];
    list.forEach(n => {
      (((n === i) || (n % i)) == 0) ? null : newList.push(n)
    });
    return this.calculateSieve(i+1, newList);
  }
}
